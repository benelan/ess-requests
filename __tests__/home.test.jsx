import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Home from '../pages/index'

describe('<Home /> snapshot', () => {
  it('matches com', () => {
    const homeTree = renderer.create(<Home />).toJSON()
    expect(homeTree).toMatchSnapshot()
  })
})

beforeEach(() => {
  render(<Home />)
})

describe('<Home /> renders', () => {
  it('title', () => {
    expect(screen.getAllByText(/ESS Requests/)).toHaveLength(2)
  })
  it('instructions', () => {
    expect(
      screen.getByText(/Which request form do you need\?/),
    ).toBeInTheDocument()
  })
})
