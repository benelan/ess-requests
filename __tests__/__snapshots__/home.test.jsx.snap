// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> snapshot matches com 1`] = `
<div>
  <title>
    ESS Requests
  </title>
  <nav
    className="navbar navbar-expand navbar-light border-bottom"
    style={
      Object {
        "background": "white",
      }
    }
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/training"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Training
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/exam"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Exam
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="container"
  >
    <div
      className="jumbotron mt-4"
      style={
        Object {
          "background": "#e1f2fb",
        }
      }
    >
      <h1
        className="display-5 text-center"
      >
        ESS Requests
      </h1>
      <p
        className="lead mt-4"
      >
        Which request form do you need?
      </p>
      <hr
        className="mb-4"
      />
      <div
        className="container-fluid content-row"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-6  my-2"
          >
            <div
              className="card h-100 shadow hoverForm"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Request for Training
                </h5>
                <p
                  className="card-text"
                >
                  When requesting instructor led or course training that requires management approval.
                </p>
              </div>
            </div>
          </div>
          <div
            className="col-sm-6 my-2"
          >
            <div
              className="card h-100 shadow hoverForm"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Request for Exam Certification
                </h5>
                <p
                  className="card-text"
                >
                  Formal request for taking a certification exam.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
